From 00c237c57a15cfb28464f97b2aefac2014b41f34 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Wed, 15 Jan 2025 16:04:29 +0100
Subject: [PATCH 1/2] tolower_and_remove_ws: always increment the src pointer

If whitespace is encountered, the function would spin indefinitely.
Skip the whitespace instead.

Upstream-Status: Submitted [https://github.com/any1/neatvnc/pull/141]
---
 src/stream/ws/handshake.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/stream/ws/handshake.c b/src/stream/ws/handshake.c
index b1b70d3..85b1b3e 100644
--- a/src/stream/ws/handshake.c
+++ b/src/stream/ws/handshake.c
@@ -29,9 +29,11 @@ static const char magic_uuid[] = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11";
 
 static void tolower_and_remove_ws(char* dst, const char* src)
 {
-	while (*src)
+	while (*src) {
 		if (!isspace(*src))
-			*dst++ = tolower(*src++);
+			*dst++ = tolower(*src);
+		src++;
+	}
 	*dst = '\0';
 }
 
-- 
2.43.0

